src/java/org/apache/cassandra/db/guardrails/EnableFlag.java

```diff
@@ -93,7 +118,16 @@ public void ensureEnabled(@Nullable ClientState state)
      */
     public void ensureEnabled(String featureName, @Nullable ClientState state)
     {
-        if (!isEnabled(state))
+        if (!enabled(state))
+            return;
+
+        if (!enabled.test(state))
+        {
             fail(featureName + " is not allowed", state);
+            return;
+        }
+
+        if (warned.test(state))
+            warn(featureName);
```

Comment:
Why not add something similar to what we have on the call to `fail`? For example:
```suggestion
            warn(featureName + " is not recommended");
```